Source/WebCore/platform/graphics/transforms/AffineTransform.cpp

```diff
@@ -349,6 +350,13 @@ Region AffineTransform::mapRegion(const Region& region) const
     return mappedRegion;
 }
 
+Path AffineTransform::mapPath(const Path& path) const
+{
+    Path transformedPath = path;
```

Comment:
I would expect a dedicated case here (similar to mapRegion / ...) to handle at least identity matrices, or even better also translations. Obviously no benefit for non-scaling-stroke transform, but mapPath() should be generic, if exposed like this.

```
if (isIdentityOrTranslation()) {
    Path mappedPath(path);
    mappedPath.translate(FloatSize(narrowPrecisionToFloat(m_transform[4], ...
    return mappedPath;
}
```

Reply:
What do you think about making Cairo's Path::translate and Path::transform check for an identity transform, rather than having that check in AffineTransform::mapPath? The CG version of Path::transform does this (and Path::translate just calls transform).

Reply:
I actually expect "mapPath" to not alter the Path by the transformation ("out-of-place"), whereas Path::tranform() indicates in-place operations to me. I don't find it too confusing, but also wouldn't mind a more friendly naming scheme, if present.

Furthermore you only looked at the CG variant of Path::transform(), which indeed reallocates (CGPathCreateMutableCopyByTransformingPath), as that's the preferred CG solution. However the Cairo variant in PathCairo.cpp, actually keeps the internal path data and transforms each coordinate. Re-creating paths in Cairo is quite heavy, so this is in fact an optimization for Cairo. Same optimization was applied for Path::translate() for Cairo platforms.

Reply:
Regarding the comment from @heycam - I think the identity check should be at the call-site, but that's my personal preference, one can potentially combine multiple such checks into one at the call-sites, where you cannot do that if the called function is not inlined...